sycl/test-e2e/KernelFusion/diamond_shape_new_driver.cpp

```diff
@@ -0,0 +1,106 @@
+// RUN: %{build} %{embed-ir} -O2 --offload-new-driver -o %t.out
```

Comment:
Can we put this in the `sycl/test-e2e/NewOffloadDriver` folder instead?

Reply:
What is the plan for these tests after the new offload driver becomes the default?

If the plan is to delete the tests using the new driver explicitly after the default behavior is changed, I agree with @sarnex that it would make sense to put them all in the same place.

Reply:
hi Lukas and Nick

Thanks so much for feedback here.
 I am going to remove the E2E test here. I have a new PR coming in very soon that adds a sequence of test for new driver and I will include this there.
I evaluated three options:
1. Adding a new file in the same location as the original test
2. Adding a new file in a separate folder
3. Adding few lines in existing test to enable new offload testing

It seems to me that Option #3 will be the easiest to maintain when we make changes to the test in the future.
@sarnex and @sommerlukas WDYT?

Thanks

Reply:
IMO 2) is the easiest to maintain, they're all in the same place and won't bog down teams not working on tools

Reply:
Hi @sarnex . I thought about that. But, users making changes to the original test should remember to change the NewOffloadModel test as well. I thought that was a painpoint.

Reply:
I don't feel super strongly on this, so if the test owners are okay with it, it's fine with me.

Reply:
I don't feel strong about the location, but I'd prefer if we had a test actually using the IR embedded by the flag. 

For example, I think that none of the tests currently in this PR check whether the embedded IR has the correct target name/prefix as expected by the SYCL runtime. 

With an e2e test, we would ensure that IR embedding is actually functional. 

Option 3 would also be viable to achieve that.

Reply:
I am adding the test back. Based on further 'internal thinking' and the fact that there is no 'strong' push for a particular option , I have decided to go with option #2. This will atleast avoid triaging headaches as @sarnex suggested.

Thanks again for discussion.